### PR TITLE
Remove balanced attached node volumes

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
https://github.com/kubernetes/kubernetes/pull/60525/ introduced
Balanced attached node volumes feature gate to include volume
count for prioritizing nodes. The reason for introducing this
flag was its usefulness in Red Hat OpenShift Online environment
which is not being used any more. So, removing the flag
as it helps in maintainability of the scheduler code base
as mentioned at https://github.com/kubernetes/kubernetes/pull/101489#discussion_r633568987
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Remove balanced attached node volumes feature
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
Remove balanced attached node volumes feature
```

### CODE REVIEW

File: pkg/scheduler/framework/interface.go [hunk 1]
```diff
@@ -478,7 +478,7 @@ type Framework interface {
 	// preemption, we may pass a copy of the original nodeInfo object that has some pods
 	// removed from it to evaluate the possibility of preempting them to
 	// schedule the target pod.
-	RunFilterPlugins(ctx context.Context, state *CycleState, pod *v1.Pod, nodeInfo *NodeInfo) PluginToStatus
+	//RunFilterPlugins(ctx context.Context, state *CycleState, pod *v1.Pod, nodeInfo *NodeInfo) PluginToStatus
```
Comment (ingvagabund) [inline pkg/scheduler/framework/interface.go:481]: Can you move removal of the methods into a separate PR? Building your changes locally without changing the `Framework` interface, I don't see any compilation errors.

**Review by ingvagabund — CHANGES_REQUESTED** [review]
Mostly removing all the blocks which are conditioned by the feature being enabled. Do you also plan to remove the feature flag itself? It's still alpha. No guarantees the flag will be there. If the featuregate for `BalanceAttachedNodeVolumes` is set but the name is not recognized, will the scheduler complain?

File: pkg/scheduler/framework/interface.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/scheduler/framework/interface.go:481]: Are these comments intentional?

**Review by alculquicondor — COMMENTED** [review]
+1 to remove the feature gate. The scheduler won't start if the feature name is not recognized, but that's expected. So the feature name should be mentioned in the release note (and perhaps add ACTION REQUIRED).

**Comment by ahg-g** [comment]
Please also remove the feature from [kube_features.go](https://github.com/kubernetes/kubernetes/blob/f31bfd8355cc1fb9cf0cb78a24f94d6d7a9012ea/pkg/features/kube_features.go#L220); and send a followup PR to cleanup the docs as well: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

File: pkg/scheduler/framework/interface.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply pkg/scheduler/framework/interface.go]: I see the following error in my GoLand IDE. While this is not causing compilation issues, I see the following:

```
Duplicate method 'RunFilterPlugins'
Duplicate method 'RunPreFilterExtensionAddPod'
Duplicate method 'RunPreFilterExtensionRemovePod'
Duplicate method 'RunPreScorePlugins'
Duplicate method 'RunScorePlugins'
```

coming from ```Handle -> PluginsRunner```
I was initially thinking it was caused by my changes but seems master is also having the same problem. Unrelated but why to have those methods implicitly and explicitly within embedded interfaces and Framework interface?

**Comment by ravisantoshgudimetla** [comment]
> Please also remove the feature from kube_features.go; and send a followup PR to cleanup the docs as well: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

Sure, I thought I pushed it the modified kube_features.go as well but missed it. I've pushed it now. I'll send a docs PR soon.

**Comment by ravisantoshgudimetla** [comment]
I think I should send an email to devel mailing list to make sure that this comes as not a surprise to users again later.

**Comment by ahg-g** [comment]
> I think I should send an email to devel mailing list to make sure that this comes as not a surprise to users again later.

sounds good, pls cc sig-scheduling mailing list as well.

File: pkg/scheduler/framework/interface.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/scheduler/framework/interface.go]: Ah... it seems like a leftover from a refactoring. I'll open a separate PR for that.

File: pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go [hunk 2]
```diff
@@ -24,114 +24,14 @@ import (
...
-		Volumes: []v1.Volume{
-			{
-				VolumeSource: v1.VolumeSource{
-					AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{VolumeID: "ovp1"},
-				},
-			},
-		},
-		NodeName: "machine4",
-	}
+
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go:34]: nit: remove empty line

**Review by alculquicondor — COMMENTED** [review]
Anything else you plan to do or can you remove the WIP from the name?

File: pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go (hunk 2 above)
Comment (ravisantoshgudimetla) [reply pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go]: Done. Let's wait for a week more and get this feature removed.

Comment (alculquicondor) [reply pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go]: Let's proceed?

Comment (ravisantoshgudimetla) [reply pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go]: +1

**Comment by alculquicondor** [comment]
This needs another rebase :(

**Review by alculquicondor — COMMENTED** [review]
/triage accepted
/lgtm
/approve
